src/lib/rng/hmac_drbg/hmac_drbg.cpp

```diff
@@ -10,35 +10,50 @@
 
 namespace Botan {
 
-HMAC_DRBG::HMAC_DRBG(MessageAuthenticationCode* hmac,
+HMAC_DRBG::HMAC_DRBG(std::unique_ptr<MessageAuthenticationCode> prf,
+                     RandomNumberGenerator& underlying_rng,
                      size_t max_output_before_reseed) :
-   Stateful_RNG(max_output_before_reseed),
-   m_mac(hmac)
+   Stateful_RNG(underlying_rng, max_output_before_reseed),
+   m_mac(std::move(prf))
    {
-   m_V.resize(m_mac->output_length());
+   BOTAN_ASSERT_NONNULL(m_mac);
    clear();
    }
 
-HMAC_DRBG::HMAC_DRBG(const std::string& hmac_hash,
+HMAC_DRBG::HMAC_DRBG(std::unique_ptr<MessageAuthenticationCode> prf,
+                     RandomNumberGenerator& underlying_rng,
+                     Entropy_Sources& entropy_sources,
                      size_t max_output_before_reseed) :
-   Stateful_RNG(max_output_before_reseed)
+   Stateful_RNG(underlying_rng, entropy_sources, max_output_before_reseed),
+   m_mac(std::move(prf))
    {
-   const std::string hmac = "HMAC(" + hmac_hash + ")";
+   BOTAN_ASSERT_NONNULL(m_mac);
+   clear();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this call from a constructor/destructor to the overridable "clear" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=cpp%3AS1699)